tests/contrib/operators/test_gcp_base.py

```diff
@@ -0,0 +1,172 @@
+        session = settings.Session()
+        try:
+            conn = session.query(models.Connection).filter(
+                models.Connection.conn_id == 'google_cloud_default')[0]
+            extras = conn.extra_dejson
+            extras[KEYPATH_EXTRA] = self.full_key_path
+            del extras[KEYFILE_DICT_EXTRA]
+            extras[SCOPE_EXTRA] = 'https://www.googleapis.com/auth/cloud-platform'
+            extras[PROJECT_EXTRA] = self.project_extra
+            conn.extra = json.dumps(extras)
+            session.commit()
+        except BaseException as e:
+            print('Airflow DB Session error:' + str(e.message))
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+    def update_connection_with_dictionary(self):
+        session = settings.Session()
+        try:
+            conn = session.query(models.Connection).filter(
+                models.Connection.conn_id == 'google_cloud_default')[0]
+            extras = conn.extra_dejson
+            with open(self.full_key_path, "r") as f:
+                content = json.load(f)
+            extras[KEYFILE_DICT_EXTRA] = json.dumps(content)
+            del extras[KEYPATH_EXTRA]
+            extras[SCOPE_EXTRA] = 'https://www.googleapis.com/auth/cloud-platform'
+            extras[PROJECT_EXTRA] = self.project_extra
+            conn.extra = json.dumps(extras)
+            session.commit()
+        except BaseException as e:
+            print('Airflow DB Session error:' + str(e.message))
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+    def _symlink_dag(self):
+        target_path = os.path.join(TESTS_DAG_FOLDER, self.dag_name)
+        if os.path.exists(target_path):
+            os.remove(target_path)
+        os.symlink(
+            os.path.join(self.example_dags_folder, self.dag_name),
+            os.path.join(target_path))
+
+    def _rm_symlink_dag(self):
+        os.remove(os.path.join(TESTS_DAG_FOLDER, self.dag_name))
+
+    def _run_dag(self):
+        dag_bag = models.DagBag(dag_folder=TESTS_DAG_FOLDER, include_examples=False)
+        self.args = {'owner': 'airflow', 'start_date': DEFAULT_DATE}
+        dag = dag_bag.get_dag(self.dag_id)
+        dag.clear(reset_dag_runs=True)
+        dag.run(ignore_first_depends_on_past=True, verbose=True)
+
+    def setUp(self):
+        configuration.conf.load_test_config()
+        self._gcp_authenticate()
+        self._symlink_dag()
+
+    def tearDown(self):
+        self._rm_symlink_dag()
+
+    @staticmethod
+    def skip_check(key):
+        if GCP_FOLDER_ENVIRONMENT_VARIABLE not in os.environ:
+            return True
+        key_folder = os.environ[GCP_FOLDER_ENVIRONMENT_VARIABLE]
+        if not os.path.isdir(key_folder):
+            return True
+        key_path = os.path.join(key_folder, key)
+        if not os.path.isfile(key_path):
+            return True
+        return False
```

Comment:
Awesome

Reply:
Yeah. As soon as it will get into master I will make an update to the proposal of Integration testing with GCP (https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-4+Support+for+Integration+Tests?focusedCommentId=93324391#comment-93324391) and will resume the conversation. I want to update all our tests in all GCP properties to use this test_gcp_base.py - it makes our testing much easier and faster.